### PR TITLE
chore(#746): Migrated docusaurus within packages.

### LINKED ISSUE #746: [Chore]: Migrate docusaurus within packages
## Is your chore related to a problem? Please describe:
Having the documentation package on the root of the monorepo introduce certain intrinsic monorepo workflow problems and also goes against the standard convention of grouping all packages together.
 
## Describe the solution you'd like:
Since the docusaurus instance already needs an update, the best course of action would be to setup a new `documentation` package within the `packages` directory and then migrate all the existing docusaurus config and documents within the new instance while preserving the git history from the older package.

### PR DESCRIPTION
**Pull request by sayak-sarkar**
## Closes / Fixes / Resolves
Resolves #746 

## Explain the feature/fix

Migrated all docusaurus content and config to the new documentation package.

## Does this PR introduce a breaking change
No

### CODE REVIEW

**Review by kunyan — APPROVED** [review]
LGFM

File: packages/documentation/package.json [hunk 1]
```diff
@@ -1,19 +1,22 @@
 {
-  "name": "docusaurus",
+  "name": "documentation",
```
Comment (mwcz) [inline packages/documentation/package.json:2]: Even though this package won't be published to npm, just for consistency with the other packages, I think this should be:

```suggestion
  "name": "@spaship/documentation",
```

Comment (sayak-sarkar) [reply packages/documentation/package.json:2]: Great idea! :+1:

**Comment by mwcz** [comment]
I think this is ready to merge, once the conflicts are resolved.

**Pull request merged**